src/redprl/syntax.sml

```diff
@@ -15,6 +17,7 @@ struct
    | DFUN of 'a * variable * 'a | LAM of variable * 'a | AP of 'a * 'a
    | DPROD of 'a * variable * 'a | PAIR of 'a * 'a | FST of 'a | SND of 'a
    | ID_TY of (symbol * 'a) * 'a * 'a | ID_ABS of symbol * 'a | ID_AP of 'a * param
+   | HCOM of param list (* extents *) * dir * 'a (* type *) * 'a (* cap *) * (symbol * 'a) list (* tubes *)
```

Comment:
one way to make this conversion lossless (I think) would be to include a boolean parameter here that says whether the hcom is canonical—which would only be meaningful assuming that the type parameter is a canonical type. Or maybe there is a cleaner way...